pkg/controller/jivavolume/jivavolume_controller.go

```diff
@@ -589,12 +591,98 @@ func updateJivaVolumeWithServiceInfo(r *ReconcileJivaVolume, cr *jv.JivaVolume,
 	return nil
 }
 
+func getBaseReplicaTolerations() []corev1.Toleration {
+	return []corev1.Toleration{
+		corev1.Toleration{
+			Key:      "node.alpha.kubernetes.io/notReady",
+			Effect:   corev1.TaintEffectNoExecute,
+			Operator: corev1.TolerationOpExists,
+		},
+		corev1.Toleration{
+			Key:      "node.cloudprovider.kubernetes.io/uninitialized",
+			Effect:   corev1.TaintEffectNoExecute,
+			Operator: corev1.TolerationOpExists,
+		},
+		corev1.Toleration{
+			Key:      "node.alpha.kubernetes.io/unreachable",
+			Effect:   corev1.TaintEffectNoExecute,
+			Operator: corev1.TolerationOpExists,
+		},
+		corev1.Toleration{
+			Key:      "node.kubernetes.io/not-ready",
+			Effect:   corev1.TaintEffectNoExecute,
+			Operator: corev1.TolerationOpExists,
+		},
+		corev1.Toleration{
+			Key:      "node.kubernetes.io/unschedulable",
+			Effect:   corev1.TaintEffectNoExecute,
+			Operator: corev1.TolerationOpExists,
+		},
+		corev1.Toleration{
+			Key:      "node.kubernetes.io/unreachable",
+			Effect:   corev1.TaintEffectNoExecute,
+			Operator: corev1.TolerationOpExists,
+		},
+		corev1.Toleration{
+			Key:      "node.kubernetes.io/out-of-disk",
+			Effect:   corev1.TaintEffectNoExecute,
+			Operator: corev1.TolerationOpExists,
+		},
+		corev1.Toleration{
+			Key:      "node.kubernetes.io/memory-pressure",
+			Effect:   corev1.TaintEffectNoExecute,
+			Operator: corev1.TolerationOpExists,
+		},
+		corev1.Toleration{
+			Key:      "node.kubernetes.io/disk-pressure",
+			Effect:   corev1.TaintEffectNoExecute,
+			Operator: corev1.TolerationOpExists,
+		},
+		corev1.Toleration{
+			Key:      "node.kubernetes.io/network-unavailable",
+			Effect:   corev1.TaintEffectNoExecute,
+			Operator: corev1.TolerationOpExists,
+		},
+	}
+}
+
+func getBaseTargetTolerations() []corev1.Toleration {
+	var zero int64
+	return []corev1.Toleration{
+		corev1.Toleration{
+			Key:               "node.alpha.kubernetes.io/notReady",
+			Effect:            corev1.TaintEffectNoExecute,
+			Operator:          corev1.TolerationOpExists,
+			TolerationSeconds: &zero,
+		},
+		corev1.Toleration{
+			Key:               "node.alpha.kubernetes.io/unreachable",
+			Effect:            corev1.TaintEffectNoExecute,
+			Operator:          corev1.TolerationOpExists,
+			TolerationSeconds: &zero,
+		},
+		corev1.Toleration{
+			Key:               "node.kubernetes.io/not-ready",
+			Effect:            corev1.TaintEffectNoExecute,
+			Operator:          corev1.TolerationOpExists,
+			TolerationSeconds: &zero,
+		},
+		corev1.Toleration{
+			Key:               "node.kubernetes.io/unreachable",
+			Effect:            corev1.TaintEffectNoExecute,
+			Operator:          corev1.TolerationOpExists,
+			TolerationSeconds: &zero,
+		},
+	}
+}
```

Comment:
@shubham14bajpai could you verify it once?
I got a feedback last time that controller didn't schedule to other node when node was in notReady state.
Additionally it would be great if you include a BDD in maya repo to verify the same with provisioning code in maya.
Thanks.

Reply:
@utkarshmani1997 I have verified this locally by running on a 4 node cluster
Tainted the node with NoExecute and the pod was evicted and moved to another node.